modules/external-auth-configuring.adoc

```diff
@@ -0,0 +1,192 @@
+// Module included in the following assemblies:
+//
+// * authentication/external-auth.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="external-auth-configuring_{context}"]
+= Configuring an OIDC identity provider for direct authentication
+
+You can configure {product-title} to directly use an external OIDC identity provider to issue tokens for authentication.
+
+:FeatureName: Direct authentication with an OIDC identity provider
+include::snippets/technology-preview.adoc[]
+
+.Prerequisites
+
+* You have enabled the `TechnologyPreviewNoUpgrade` feature set.
+* You have configured your external authentication provider. This procedure uses Keycloak as the identity provider and assumes that you have the following clients configured:
+
+** A client for the web console called `console-test`
++
+// TODO: Should we mention anything about this being an authenticated client?
+** A client for the {oc-first} called `oc-cli-test`
+* You have access to the `kubeconfig` file generated by the installation program for the cluster.
++
+[IMPORTANT]
+====
+Before replacing the built-in OAuth server with an external provider, ensure that you have access to a valid `kubeconfig` file.
+
+If there are any issues with the external identity provider, you can use the `kubeconfig` file to gain access to the {product-title} cluster in an emergency situation.
+====
+
+.Procedure
+
+. Log in to the cluster by using your `kubeconfig` file.
++
+You must use the `kubeconfig` file to configure the direct authentication. If you use a token, the token can become invalid during the process of the procedure.
+
+. Create a secret that allows you to authenticate with the web console by running the following command:
++
+[source,terminal]
+----
+$ oc create secret generic console-secret \
+    --from-literal=clientSecret=<secret_value> \// <1>
+    -n openshift-config
+----
+<1> Replace `<secret_value>` with the value of the secret for the `console-test` client in Keycloak.
++
+////
+TODO: I skipped a step that did this:
```

Comment:
@liouk @xingxingxia I skipped the part that extracted the ca-bundle.crt from the cluster, because I'm assuming we only did that because we were running keycloak on the same cluster. Can you confirm that we don't need to do that, and that I'm correct in how I worded the next step about creating the config map and providing their provider's ca-bundle.crt?

Reply:
@bergerhoffer , correct